### ISSUE TITLE
2.4.2 Admin UI says upgrade available even when on the latest version

### ISSUE THREAD

**Issue by celogeek** [rethinkdb/rethinkdb#7072]
This is the same issue as here https://github.com/rethinkdb/rethinkdb/issues/6910
I'm trying to understand how to generate again the asset with the right version but no success so far.

**Comment by srh** [comment]
You can generate it by checking out https://github.com/rethinkdb/rethinkdb/tree/old_admin and following the instructions in the readme.

*However*, the PVERSION parameter seems to be ignored.  There is a bug in `make generate` or some script that it calls.  It might be possible to work around this by setting the git tag you want on the current commit, and generating then.

**Comment by celogeek** [comment]
Well, the issue is on the release, even on the debian package you also have the issue.
When we open the old admin ui, then we have a pop up indicating a fresher version is available despite the fact that we are using it.
may be the process to embed or release should include the generation of the asset with the right version.

**Comment by celogeek** [comment]
I will take a look, but I really doubt I should do this extract step in the build of the server only.

**Comment by celogeek** [comment]
I try to build the admin UI, but it loop forever.
I think the node6 (which is very outdated), can't build on my setup.
I know I have to apply 2 patches before to make it works
```
--- mk/support/pkg/v8.sh
+++ mk/support/pkg/v8.sh
@@ -155,7 +155,7 @@
         if [[ "$arch" = "ppc64" ]]; then
             arch_gypflags="$arch_gypflags -Duse_system_icu=1"
         fi
-        pkg_make $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
+        pkg_make $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= -Dv8_use_snapshot=false $arch_gypflags" V=1
         for lib in `find "$build_dir/out/$arch.$mode" -maxdepth 1 -name \*.a` `find "$build_dir/out/$arch.$mode/obj.target" -name \*.a`; do
             name=`basename $lib`
             cp $lib "$install_dir/lib/${name/.$arch/}"
@@ -168,7 +168,7 @@
     # These are the necessary libraries recommended by the docs:
     # https://developers.google.com/v8/get_started#hello
      if [[ "$ARCH" != "ppc64le" ]]; then
-        for lib in libv8_{base,libbase,snapshot,libplatform}; do
+        for lib in libv8_{base,libbase,nosnapshot,libplatform}; do
              echo "$install_dir/lib/$lib.a"
         done
         for lib in libicu{i18n,uc,data}; do
```

and

```
--- deps/v8_inspector/third_party/v8_inspector/platform/v8_inspector/String16STL.cpp.old        2021-09-08 18:09:36.288899969 +0200
+++ deps/v8_inspector/third_party/v8_inspector/platform/v8_inspector/String16STL.cpp    2021-09-08 18:10:06.528708959 +0200
@@ -4,6 +4,7 @@

 #include "platform/inspector_protocol/InspectorProtocol.h"

+#include <limits>
 #include <algorithm>
 #include <cctype>
 #include <cstdio>
```


This is because my gcc is too recent for this node version.
Did you build this ui on an old machines ?

**Comment by srh** [comment]
Branch https://github.com/rethinkdb/rethinkdb/tree/old_admin doesn't have a file mk/support/pkg/v8.sh.  So you are not generating the admin UI on the latest commit in that branch.

**Comment by vincent-ogury** [comment]
I've got those patch on the previous version of rethink.
I've try to build the old admin with the last patch, and failed.
I'm on arch linux with everything up to date (gcc, clang ...).
It seems node 6 failed to build without patches there.

**Comment by celogeek** [comment]
I've got those patch on the previous version of rethink.
I've try to build the old admin with the last patch, and failed.
I'm on arch linux with everything up to date (gcc, clang ...).
It seems node 6 failed to build without patches there.

**Comment by srh** [comment]
From #6910 :
 
> We could remove this step if the web UI used r.db('rethinkdb').table('server_status') to retrieve the server version.

**Comment by srh** [comment]
I've retitled this and have pinned this issue.

**Comment by srh** [comment]
This is now fixed in the `v2.4.x` branch (as the 2.4.3-version-numbered web assets has been properly generated in preparation for the 2.4.3 release).  I'm closing, but keeping this issue pinned.
